src/libraries/QuoterRevert.sol

```diff
@@ -28,28 +30,20 @@ library QuoterRevert {
         }
     }
 
-    /// @notice check revert reasons are of the expected length; otherwise revert with different message
-    /// @dev called after a swap simulation reverts, to check if the revert was valid encoded quote information, or an internal issue
-    function validateRevertReason(bytes memory reason) internal pure {
+    /// @notice validates whether a revert reason is a valid swap quote or not
+    /// if valid, it decodes the quote to return. Otherwise it reverts.
+    function parseQuoteAmount(bytes memory reason) internal pure returns (uint256 quoteAmount) {
+        // If the error doesnt start with QuoteSwap, we know this isnt a valid quote to parse
+        // Instead it is another revert that was triggered somewhere in the simulation
         if (reason.parseSelector() != QuoteSwap.selector) {
             revert UnexpectedRevertBytes(reason);
         }
-    }
-
-    /// @notice validates a received revert reason from a swap.
-    /// Then, if valid, decodes it into the information generated by a quote
-    function parseReturnData(bytes memory reason) internal pure returns (uint256) {
-        reason.validateRevertReason();
-        return reason.parseAmountUnspecified();
-    }
 
-    /// @notice extracts the amountUnspecified from an encoded QuoteSwap(amountUnspecified)
-    function parseAmountUnspecified(bytes memory reason) internal pure returns (uint256 amountUnspecified) {
         // reason -> reason+0x1f is the length of the reason string
         // reason+0x20 -> reason+0x23 is the selector of QuoteSwap
-        // reason+0x24 -> reason+0x43 is the amountUnspecified
+        // reason+0x24 -> reason+0x43 is the quoteAmount
```

Comment:
I guess this is more of a comment to all of this natspec but this is kinda confusing.

0x20 -> 0x23 is a 3 byte difference?
0x24 -> 0x43 is a 31 byte difference? 

Is something not inclusive?

Reply:
it is inclusive thats why... 0x20, 0x21, 0x22, 0x23 is 4 bytes inclusively